### PR TITLE
feat: Issues tab view

### PR DESCRIPTION
**Pull request by markshenouda**
image

### CODE REVIEW

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
+// I must disable eslint because otherwise everytime I save the file
+// elsint brokes my class declaration at line #38 and the icon hasn't the right color
+
+/* eslint-disable */
```
Comment (danecando) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx:4]: Can you just disable it for that line? 
`// eslint-disable-next-line`

**Review by danecando — DISMISSED** [review]
👍

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx (hunk 1 above)
Comment (markshenouda) [reply solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx]: I solved the main issue so I removed it as well

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesData/PRAndIssuesData.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        each={
+          activeTab() === 'OPEN'
+            ? issuesStore().openIssues.issues
+            : issuesStore().closedIssues.issues
+        }
+      >
+        {(issue) => <PRAndIssuesListItem {...issue} />}
+      </For>
+
+      {activeTab() === 'OPEN' && issuesStore().openIssues.totalCount === 0 && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesData/PRAndIssuesData.tsx:21]: use Switch and Match here, please or Show.

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesData/PRAndIssuesData.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        <NoResults />
+      )}
+
+      {activeTab() === 'CLOSED' &&
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesData/PRAndIssuesData.tsx:25]: same as above. use Show

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesHeader/PRAndIssuesHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  return (
+    <div class="flex flex-wrap space-x-1 space-y-2 md:space-x-0 md:space-y-0 items-center justify-between p-4 bg-gray-100 border-b rounded-t-lg">
+      <div class="flex space-x-4">
+        <button
+          class={cn('text-xs flex items-center gap-1 text-gray-600', {
+            'font-semibold text-gray-900': activeTab() === 'OPEN',
+          })}
+          onClick={() => setActiveTab('OPEN')}
+        >
+          {type() === 'pr' ? (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesHeader/PRAndIssuesHeader.tsx:62]: use Switch and Match here

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesHeader/PRAndIssuesHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+          })}
+          onClick={() => setActiveTab('CLOSED')}
+        >
+          <CheckIcon class="w-4 h-4" />
+          <span>{issuesStore().closedIssues?.totalCount || 0}</span>
+          Closed
+        </button>
+      </div>
+      <div class="flex items-center space-x-8">
+        {labelOptions && labelOptions()?.length !== 0 && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesHeader/PRAndIssuesHeader.tsx:82]: use Show here

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesHeader/PRAndIssuesHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            <FilterDropdown
+              name="Label"
+              selected={selectedLabel()}
+              items={labelOptions()}
+              selectOption={selectLabel}
+              class="border-none text-sm inline-flex w-full justify-center items-center gap-2"
+            />
+          </div>
+        )}
+        {milestoneOptions && milestoneOptions()?.length !== 0 && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesHeader/PRAndIssuesHeader.tsx:93]: use Show here as well

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            class={cn({
+              'text-green-600': local.state === 'OPEN',
+              'text-gray-500': local.state !== 'OPEN' && type() === 'issue',
+              'text-purple-600':
+                (local.state !== 'OPEN' && local.state === 'MERGED') ||
+                (local.state === 'CLOSED' && type() === 'issue'),
+              'text-red-600': type() === 'pr' && local.state === 'CLOSED',
+            })}
+          >
+            {type() === 'issue' ? (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx:45]: use Switch and Match here as well

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+      <div class="flex-auto min-width-0 pr-3 pl-2">
+        <div>
+          <a
+            class="align-middle no-underline markdown-title font-semibold"
+            target="_blank"
+            href={local.url}
+          >
+            {local.title}
+          </a>
+          {local.labels?.map((label) => (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx:77]: use For here

File: solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            <a href="#">{local.login}</a> was{' '}
+            {local.state === 'OPEN' ? 'open' : 'closed'} on{' '}
+            {format(new Date(local.createdAt), 'MMM d, yyyy')}
+          </span>
+        </div>
+      </div>
+
+      <div class="flex-shrink-0 w-1/5 text-right pr-3 flex-nowrap flex">
+        <span class="ml-2 pt-1 flex-1 flex-shrink-0">
+          {local.commentCount > 0 && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/PRAndIssuesListItem/PRAndIssuesListItem.tsx:102]: use Show here as well

File: solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+
+  const clearSortAndFilter = () => {
+    setSortBy('Newest');
+    setSelectedLabel(undefined);
+    setSelectedMilestone(undefined);
+  };
+
+  return (
+    <div class="md:py-12 max-w-screen-xl mx-auto">
+      {(selectedLabel() || sortBy() !== 'Newest') && (
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx:26]: use Show here as well

File: solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+        >
+          <span class="text-white rounded-md bg-gray-500 w-4 h-4">
+            <CloseIcon />
+          </span>
+          Clear filter
+        </div>
+      )}
+      <PRAndIssuesData />
+
+      {activeTab() === 'OPEN'
```
Comment (hdJerry) [inline solidstart-tanstackquery-tailwind-modules/src/components/RepoIssues/RepoIssues.tsx:39]: use Switch and Match here as well

**Review by hdJerry — CHANGES_REQUESTED** [review]
There are very small fixes as we intend to make it more of solidjs than react.

Thanks sir

**Review by hdJerry — APPROVED** [review]
Thanks, sir for making it more Solid!!

**Pull request merged**